providers/account.rb

```diff
@@ -169,7 +169,24 @@ def home_ssh_dir_resource(exec_action)
 
 def authorized_keys_resource(exec_action)
   # avoid variable scoping issues in resource block
-  ssh_keys = Array(new_resource.ssh_keys)
+  ssh_keys = []
+  Array(new_resource.ssh_keys).each do |item|
+    # if item is a valid ssh key, copy it.
+    if item =~ /^(ssh-(dss|rsa|ed25519)|ecdsa-sha2-\w+) AAAA/
```

Comment:
I think that makes sense to me. If I understand it correctly these lines would match:

```
# test line 1
 # test line 2
ecdsa-sha2-nistp256 AAAA...
environment="hi there" ssh-ed25519 AAAA...
```